app/controllers/evidence_hub_controller.rb

```diff
@@ -2,17 +2,44 @@ class EvidenceHubController < ApplicationController
   DOCUMENT_TYPES = ['Insight'].freeze
 
   def index
+    @search_form = EvidenceSummarySearchForm.new(search_form_params)
+
     documents = Mas::Cms::Document.all(
-      params: search_params
+      params: parse_filter_params
     )
     @evidence_summaries = EvidenceSummary.map(documents)
   end
 
   private
 
-  def search_params
-    { document_type: DOCUMENT_TYPES }.tap do |hash|
-      hash[:keyword] = params.require(:keyword) if params[:keyword].present?
+  def search_form_params
+    {
+      document_type: DOCUMENT_TYPES
+    }.merge(evidence_summary_search_form_params)
+  end
+
+  def evidence_summary_search_form_params
+    params.fetch(:evidence_summary_search_form, {}).permit(
+      :keyword,
+      :years_of_publication,
+      client_groups: [],
+      topics: [],
+      countries_of_delivery: []
+    )
+  end
+
+  def filter_params
+    evidence_summary_search_form_params.select do |k, _|
+      filter_keys.include?(k)
```

Comment:
I don't understand the purpose of **this** method neither the **filter keys**. Can you clarify, please? Can the FilterParser not just exclude the filters which are empty? That would be more simple.

Reply:
**filter_keys** returns an array of the filter types - client_groups, topics etc. It gets these values from the translation file via the EvidenceSummarySearchForm instance.

**filter_params** returns a subset of the **evidence_summary_search_form params** as only the filter_params need to be formatted to the `blocks` hash required by the cms api. There are no 'empty' filters. If at least one checkbox is not checked for a filter type, that filter type is not passed to the controller by the form

Still to be completed - merging the formatted filter params, or blocks hash which the FilterParser returns, with the other params i.e. document_type and keyword.